src/components/Input/SwitchField/index.tsx

```diff
@@ -1,15 +1,40 @@
 import { Field } from "react-final-form";
 import { FormGroup, Input, Label } from "reactstrap";
 
+type viewOnly = {
+    checked: boolean;
+};
+
 type Props = {
     id: string;
     label: string;
     disabled?: boolean;
     onChange?: (value: boolean) => void;
+    viewOnly?: viewOnly;
 };
 
-export default function SwitchField({ id, label, onChange = undefined, disabled = false }: Props) {
-    return (
+export default function SwitchField({ id, label, onChange = undefined, disabled = false, viewOnly }: Props) {
+    return !viewOnly ? (
+        <Field name={id} type={"checkbox"}>
+            {({ input }) => (
+                <FormGroup switch>
+                    <Input
+                        {...input}
+                        type="switch"
+                        id={id}
+                        disabled={disabled}
+                        onChange={(e) => {
+                            input.onChange(e);
+                            if (onChange) {
+                                onChange(e.target.checked);
+                            }
+                        }}
+                    />
+                    <Label for={id}>{label}</Label>
+                </FormGroup>
+            )}
+        </Field>
+    ) : (
         <Field name={id} type={"checkbox"}>
```

Comment:
I guess you can remove this Field too (and maybe BootstrapForm from the detail page too)

Reply:
I removed the Field tag and also the BootstrapForm from the detail page.